### ISSUE TITLE
[2.7][Form][choice] add doc about the new option: choice_translation_domain

### ISSUE THREAD

**Issue by aitboudad** [symfony/symfony-docs#5134]
https://github.com/symfony/symfony/pull/13651

choice_translation_domain (bool | null | string)
* true: reuse current translation_domain
* false: disabled
* null: parents translation_domain or default domain (messages)
* string: explicit domain

**Comment by WouterJ** [comment]
This has to be documented in `reference/configuration/framework` using the same format as the other options and should also have a `.. versionadded:: 2.7` directive indicating that this option was introduced in Symfony 2.7.

**Comment by nothingaa** [comment]
Is there a way to impact the whole entity in a single call?
I have tried to call the following funciton and it is not working.

```
public function configureOptions(OptionsResolver $resolver)
{
    $resolver->setDefaults(array(
        'choice_translation_domain' => true,
    ));
}
```

**Comment by ricohumme** [comment]
On what form field types is the `choice_translation_domain` effective?
I'm testing with field type `date` and it throws an error saying it's a non-existent option.
As for the docs, it has no reference yet on the use of this options.

Could some clarify?

**Comment by xabbuh** [comment]
@aitboudad Can you confirm that what @ricohumme describes is the intended behaviour?

**Comment by xabbuh** [comment]
By the way, there is an open pull request: see #5614

**Comment by aitboudad** [comment]
Sorry I don't have much time these days ^^ below how it work:
The `choice_translation_domain` is related to `translation_domain` only it's value's is `true` or `null`, it was introduced to enable/disable translation for [optgroup](https://github.com/symfony/symfony/blob/2.8/src/Symfony/Bridge/Twig/Resources/views/Form/form_div_layout.html.twig#L77) and [option](https://github.com/symfony/symfony/blob/2.8/src/Symfony/Bridge/Twig/Resources/views/Form/form_div_layout.html.twig#L83) in [Choice Fields](http://symfony.com/doc/current/reference/forms/types.html#choice-fields):

* Default value: `true`
  - choice

* Default value: `false`
  - country
  - currency
  - language
  - locale
  - timezone
  - entity

for the `date` type currently it's not possible but it can be introduced in `2.8` once https://github.com/symfony/symfony/pull/15301 got merged.

**Comment by aitboudad** [comment]
@ricohumme not sure if I understood the `Edit 2:`  can you leave us an example to reproduce it.

**Comment by ricohumme** [comment]
Sure!
When using the following code:
```
$builder->add('purchaseDate', 'date', [
            'label' => 'rma_form.label.purchaseDate',
            'placeholder' => array('year' => 'Year', 'month' => 'Month', 'day' => 'Day'),
            'widget' => 'choice',
            'required' => true,
            'translation_domain' => false
        ]);
```
The output on the screen renders the widgets without translating the contents of the widgets, however the label also shows rma_form.label.purchaseDate (not translated)

When using the following code:
```
$builder->add('purchaseDate', 'date', [
            'label' => 'rma_form.label.purchaseDate',
            'placeholder' => array('year' => 'label.year', 'month' => 'label.month', 'day' => 'label.day'),
            'widget' => 'choice',
            'required' => true
        ]);
```
The output on the screen renders the widgets and translates the content: So the placeholders are correct, but also the numbers are translated. The label widget label is correctly translated.

**Comment by aitboudad** [comment]
@ricohumme your example work as expected :), as I already mentioned above the `choice_translation_domain` doesn't affect the `date` type.

**Comment by ricohumme** [comment]
@aitboudad that is true, you did. But do you agree that the `date` type should work the same way the `choice` type does when it comes to `choice_translation_domain` ?

**Comment by aitboudad** [comment]
@ricohumme yes, see  symfony/symfony#15301

**Comment by xabbuh** [comment]
@ricohumme @aitboudad What about the label not being translated? Is that intended too?

**Comment by aitboudad** [comment]
@xabbuh when `translation_domain` = false the label is not translated see:

[{{ translation_domain is sameas(false) ? label : label|trans({}, translation_domain) }}](https://github.com/symfony/symfony/blob/2.8/src/Symfony/Bridge/Twig/Resources/views/Form/form_div_layout.html.twig#L228)

**Comment by xabbuh** [comment]
@aitboudad Yes, but was that really intended?

**Comment by aitboudad** [comment]
@xabbuh not sure, anyway not related to this issue !

**Comment by peterrehm** [comment]
Fixed by #5906
